deepchecks/vision/datasets/classification/mnist.py

```diff
@@ -205,9 +215,111 @@ def __call__(self, batch):
+        'http://yann.lecun.com/exdb/mnist/',
+        'https://ossci-datasets.s3.amazonaws.com/mnist/',
+    ]
+
+    resources = [
+        ('train-images-idx3-ubyte.gz', 'f68b3c2dcbeaaa9fbdd348bbdeb94873'),
+        ('train-labels-idx1-ubyte.gz', 'd53e105ee54ea40749a09fcbcd1e9432'),
+        ('t10k-images-idx3-ubyte.gz', '9fb629c4189551a2d022fa330f9573f3'),
+        ('t10k-labels-idx1-ubyte.gz', 'ec29112dd5afa0611ce80d1b7f02629c')]
+
+    def __init__(self, batch_size: int = 64, train: bool = True, transform: t.Optional[t.Callable] = None,
+                 n_samples: int = None) -> None:
+        super().__init__()
+        self.train = train  # training set or test set
+        self.transform = transform
+        self.batch_size = batch_size
+
+        self.download()
+        if not self._check_exists():
+            raise RuntimeError('Dataset not found.' +
+                               ' You can use download=True to download it')
+
+        self.data, self.targets = self._load_data()
+        if n_samples:
+            self.data = self.data[:n_samples]
+            self.targets = self.targets[:n_samples]
+        if self.transform is not None:
+            self.data = torch.stack([self.transform(image=img.numpy())['image'] for img in self.data])
+
+    def __iter__(self):
+        """Iterate over the dataset."""
+        return cycle(zip(self.data, self.targets))
+
+    def _load_data(self):
+        image_file = f"{'train' if self.train else 't10k'}-images-idx3-ubyte"
+        data = read_image_file(os.path.join(self.raw_folder, image_file))
+
+        label_file = f"{'train' if self.train else 't10k'}-labels-idx1-ubyte"
+        targets = read_label_file(os.path.join(self.raw_folder, label_file))
+
+        return data, targets
+
+    @property
+    def raw_folder(self) -> str:
+        """Return the path to the raw data folder."""
+        return os.path.join(MNIST_DIR, 'raw_data')
+
+    def _check_exists(self) -> bool:
+        return all(
+            check_integrity(os.path.join(self.raw_folder, os.path.splitext(os.path.basename(url))[0]))
+            for url, _ in self.resources)
+
+    def download(self) -> None:
+        """Download the MNIST data if it doesn't exist already."""
+        if self._check_exists():
+            return
+
+        os.makedirs(MNIST_DIR, exist_ok=True)
+
+        # download files
+        for filename, md5 in self.resources:
+            for mirror in self.mirrors:
+                url = f'{mirror}{filename}'
+                try:
+                    print(f'Downloading {url}')
+                    download_and_extract_archive(
+                        url, download_root=self.raw_folder,
+                        filename=filename,
+                        md5=md5
+                    )
+                except URLError as error:
+                    print(f'Failed to download (trying next):\n{error}')
+                    continue
+                finally:
+                    print()
```

Comment:
what

Reply:
was there originally, just changed " to ' to make pylint happy

Reply:
but removed the finally anyway